pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -421,6 +430,147 @@ func consumerRefMatches(consumer *corev1.ObjectReference, machine *machinev1.Mac
 	return true
 }
 
+// updateUserData fetches full ignition from MCS and updates the UserData secret reference
+func (a *Actuator) updateUserData(ctx context.Context, config *bmv1alpha1.BareMetalMachineProviderSpec,
+	machine *machinev1.Machine) error {
+
+	// Try to fetch the full ignition secret to see if it exists
+	machineUserDataSecret := &corev1.Secret{}
+	key := client.ObjectKey{
+		Name:      machine.Name + "-user-data",
+		Namespace: machine.Namespace,
+	}
+	// Secret already exists. Set secretReference and return
+	err := a.client.Get(ctx, key, machineUserDataSecret)
+	if err == nil {
+		// Set Machine's secretreference
+		config.UserData = &corev1.SecretReference{
+			Name:      machine.Name + "-user-data",
+			Namespace: machine.Namespace,
+		}
+		return nil
+	}
+
+	// userDataSecret is the pointer ignition originally
+	// created by openshift installer. Fetch it to extract
+	// its content (an URL containg full ignition and a
+	// CA cert for the URL's https connection)
+	userDataSecret := corev1.Secret{}
+	key = client.ObjectKey{
+		Name:      config.UserData.Name,
+		Namespace: config.UserData.Namespace,
+	}
+	err = a.client.Get(ctx, key, &userDataSecret)
+	if err != nil {
+		log.Printf("Cannot get %s secret: %s", config.UserData.Name, err.Error())
+		return err
+	}
+	encodedUserDataBytes := userDataSecret.Data["userData"]
+	var userData []byte
+	userData, err = base64.StdEncoding.DecodeString(string(encodedUserDataBytes))
+	if err != nil {
+		log.Printf("Error encoding userData into B64 : %s", err.Error())
+		return err
+	}
+
+	var ignConf map[string]interface{}
+	if err := json.Unmarshal(userData, &ignConf); err != nil {
+		log.Printf(string(userData))
+		log.Printf("Cannot unmarshal ignition config from userData : %s", err.Error())
+		return err
+	}
+
+	ignitionURL := ignConf["ignition"].(map[string]interface{})["config"].(map[string]interface{})["append"].([]interface{})[0].(map[string]interface{})["source"].(string)
+	caCertRaw := ignConf["ignition"].(map[string]interface{})["security"].(map[string]interface{})["tls"].(map[string]interface{})["certificateAuthorities"].([]interface{})[0].(map[string]interface{})["source"].(string)
+	caCertB64 := strings.TrimPrefix(caCertRaw, "data:text/plain;charset=utf-8;base64,")
+
+	// We didn't get any URL
+	if ignitionURL == "" {
+		err := goerrors.New("Error extracting full Ignition URL")
```

Comment:
`fmt.Error` and `fmt.Errorf` could be used instead. 

But regardless, go errors shouldn't be capitalized: https://github.com/golang/go/wiki/Errors

> Error strings should not start with a capital letter because they'll often be prefixed before printing.

Reply:
+1. Removed goerrors and just started using fmt.Errorf. Thanks.